### ISSUE TITLE
Pods don't have internet access

### ISSUE THREAD

**Issue by juansalvatella** [ubuntu/microk8s#2283]
I have a single node cluster running on a CentOS 8.3.2011 machine. None of the pods are able to ping ips or resolve dns requests. The coredns kube-system pod is able to receive requests but isn't able to resolve them:

```
INFO] 10.1.102.150:46405 - 38385 "A IN google.es. udp 27 false 512" NOERROR - 0 0.000345691s
[ERROR] plugin/errors: 2 google.es. A: read udp 10.1.102.153:58843->192.168.2.1:53: read: no route to host
```

[inspection-report-20210523_235942.tar.gz](https://github.com/ubuntu/microk8s/files/6528977/inspection-report-20210523_235942.tar.gz)

cat /proc/sys/net/ipv4/ip_forward is 1

sudo iptables -P FORWARD ACCEPT has no effect

I've tried following the steps explained here https://github.com/ubuntu/microk8s/issues/1484 but it's still not working. Any idea on how to troubleshoot futher this issue?

**Comment by balchua** [comment]
Hi,

Thanks for using MicroK8s.  
I see this in the logs.

```shell
2021-05-23 21:59:32.783 [ERROR][36] health.go 246: Health endpoint failed, trying to restart it... error=listen tcp 127.0.0.1:9099: bind: address already in use

```
I checked that you have 2 `calico-node` process running i can see from the `ps` logs. I dont know which one is the one from MicroK8s.
That might be causing the issue?

**Comment by juansalvatella** [comment]
I had a look at that and maybe the process was left running from another installation? I killed the process and restarted the microk8s cluster. I don't see that error message anymore in the logs but it's still not working.

I configured the cluster enabling storange and the dns module pointing to 192.168.2.1.  As you can see, from the host this is working as expected. Regardless of this configuration, if I set the dns module pointing to the default Google DNS servers it doesn't work either.

```
$ nslookup google.com 192.168.2.1

Server:		192.168.2.1
Address:	192.168.2.1#53

Non-authoritative answer:
Name:	google.com
Address: 142.250.200.142
Name:	google.com
Address: 2a00:1450:4003:80f::200e
```

I attach the report again.
[inspection-report-20210525_073513.tar.gz](https://github.com/ubuntu/microk8s/files/6536762/inspection-report-20210525_073513.tar.gz)

Thanks balchua for your help.

**Comment by balchua** [comment]
@juansalvatella i can see that the system is using `iptables-nft` instead of `iptables-legacy`.
Im not super familiar with all these... I guess if you can try enabling `iptables-legacy`?

**Comment by juansalvatella** [comment]
Thank you! You gave me the right hint to find the problem. The problem was that the firewall was enabled by default in Centos 8 and that's apparently a problem.

I wanted to set iptables-legacy as the default iptables but Centos 8 doesn't have the binaries. While researching if anyone had a similar issue I found this issue https://github.com/rancher/rke/issues/2432. I disabled the firewall following this instructions: https://linuxhint.com/disable-firewall-centos-8/ and now everything is working.

`sudo systemctl disable firewalld`

Thank you very much!
